internal/injector/builtin/yaml/go-main.yml

```diff
@@ -19,15 +19,27 @@ aspects:
               - signature: {}
     advice:
       - inject-declarations:
+          links:
+            - gopkg.in/DataDog/dd-trace-go.v1/internal/civisibility/integrations/gotesting
           imports:
             tracer: gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
+            testdeps: testing/internal/testdeps
           # Note: it is valid to have multiple func init() in a single compile unit (e.g, `.go` file), in which case
           # they get executed in declaration order. This means it's okay for us to add a new init function if there is
           # already one in the file, but as it currently is appended AFTER all other declarations in the file, it means
           # that it will be executed last (tracing contents of previous init functions will not be possible).
           template: |-
+            //go:linkname __dd_civisibility_isCiVisibilityEnabled gopkg.in/DataDog/dd-trace-go.v1/internal/civisibility/integrations/gotesting.isCiVisibilityEnabled
+            func __dd_civisibility_isCiVisibilityEnabled() bool
+            
             func init() {
-              tracer.Start(tracer.WithOrchestrion(map[string]string{"version": {{printf "%q" Version}}}))
+              // Only initialize if is not a test process (testdeps.ImportPath is set only on test processes since go1.9 in the first init function)
+              // and if CI Visibility has not been disabled (by the kill switch).
+              // For a test process the ci visibility instrumentation will initialize the tracer
+              isATestProcess := len(testdeps.ImportPath) != 0
```

Comment:
Im' not extremely fond of this (but if there is no other ways...).

What catches my attention is this section of the `go list` documentation:
> The -test flag causes list to report not only the named packages
but also their test binaries (for packages with tests), to convey to
source code analysis tools exactly how test binaries are constructed.
The reported import path for a test binary is the import path of
the package followed by a ".test" suffix, as in "math/rand.test".
When building a test, it is sometimes necessary to rebuild certain
dependencies specially for that test (most commonly the tested
package itself). The reported import path of a package recompiled
for a particular test binary is followed by a space and the name of
the test binary in brackets, as in "math/rand [math/rand.test]"
or "regexp [sort.test]". The ForTest field is also set to the name
of the package being tested ("math/rand" or "sort" in the previous
examples).

Hence in that cas I wonder if the toolchain even signals that `main` function as being in a `main` package in that case?

Reply:
I did change the approach to use the new [`testing.Testing()` api ](https://github.com/golang/go/commit/7f38067acb738c43d870400dd648662d31456f5f#diff-b3e6126779b5ec9d3d6cea7cc54054ba78f4d7a0a6248d9e458bbd9b3d72fce3R659)added in Go 1.21 on [bedccb0](https://github.com/DataDog/orchestrion/pull/340/commits/bedccb03e7e5586f4e3256415c113fbe160a01f5)